# Segundo Examen Parcial
## Probabilidad y Estadística

Legajo a2412

Quiroga, Martin Gabriel

### Ejercicio 1
(3 puntos) Una variable aleatoria discreta X puede tomar los valores 0, 1, 2 y 3. Las probabilidades para cada valor posible
están dadas por la siguiente tabla:
|  X  |  0  |  1  |  2  |  3  |
| --- | --- | --- | --- | --- |
| p | $\frac{1\theta}{3}$ | $\frac{2\theta}{3}$ | $\frac{1 - 1\theta}{3}$ | $\frac{2\left(1 - 1\theta\right)}{3}$ |

Si experimentalmente se obtienen los siguientes datos:
(0,3,3,1,1,3,3,0,2,2),
determine el valor de $\theta$ usando el método de máxima verosimilitud.

La f.d.p. de la muestra es una función $f(x, \theta)$ dada por la tabla de la consigna. Tomando el parámetro $\theta$ como una variable y a las variables $X$ como independientes, la densidad conjunta de la muestra puede verse como una función sobre $\theta$:
$$ L(\theta) = f_{X_1, X_2, ..., X_n}(x_1, x_2, ..., x_n, \theta) = f_{X_1}(x_1, \theta) \cdot f_{X_2}(x_2, \theta) \cdots f_{X_n}(x_n, \theta) = \prod_{i=1}^n f_{x_i}(x_i, \theta)
$$

Lo que nos sirve para obtener la función de verosimilitud para $\theta$. Reemplazando, se obtiene:
$$
L(\theta) = \left(\frac{1\theta}{3}\right)^2 \left(\frac{2\theta}{3}\right)^2 \left(\frac{1 - 1\theta}{3}\right)^2 \left(\frac{2\left(1 - 1\theta\right)}{3}\right)^4
$$
$$L(\theta) = \frac{\theta^2}{3^2} \frac{2^2\theta^2}{3^2} \frac{(1-\theta)^2}{3^2} \frac{2^4(1-\theta)^4}{3^4} = \frac{4\theta^4 16(1-\theta)^6}{3^{10}}$$
$$ L(\theta) = \frac{64}{3^{10}}\theta^4(1 - \theta)^6
$$

Ahora se debe obtener el valor de $\theta$ que maximice la verosimilitud y esto se puede hacer derivando $L(\theta)$ e igualando a 0, para despejar. Si bien la fórmula de $L(\theta)$ no es extremadamente compleja de derivar, ya que termina siendo un polinomio, iguamente se puede aprovechar que al ser el logaritmo una función monótona creciente, se encontrará el mismo argumento que maximiza la verosimilitud pero de una manera mucho más sencilla matemáticamente.
$$
l(\theta) = \ln L(\theta) = \ln \left(\frac{64}{3^{10}}\theta^4(1 - \theta)^6\right) = \ln\left(\frac{64}{3^{10}}\right) + 4 \ln \theta + 6 \ln(1 - \theta)
$$

Derivando e igualando a 0:
$$
\frac{d l(\theta)}{d \theta} = 4 \frac{1}{\theta} + 6 \left(-\frac{1}{1-\theta}\right) = \frac{4}{\theta} - \frac{6}{1-\theta} = 0
$$

Despejando:
$$\frac{4}{\theta} = \frac{6}{1-\theta}$$
$$4 (1 - \theta) = 6\theta$$
$$4 - 4\theta = 6\theta$$
$$4 = 10\theta$$
$$\theta = \frac{2}{5} = 0.4$$

Entonces, el valor de $\theta$ que maximiza la verosimilitud para los datos observados es $0.4$.

### Ejercicio 2
(3 puntos) Se pretende estimar los valores de producción Y (en miles de toneladas) de cierto material, en función del tiempo transcurrido X (en meses) usando los valores de la tabla:
| X | Y |
| -- | -- |
| 0 | 2 |
| 6 | 72 |
| 10 | 137 |
| 16 | 271 |
| 21 | 532 |

Se plantea un modelo de la forma $Y = a + b x + cx^2$. Encontrar los estimadores de mínimos cuadrados para $a$, $b$ y $c$ en este modelo.

Se busca aproximar la variable aleatoria $Y$, mediante una función $\tilde{Y}$, calculada en $X$ que minimice el error cuadrático medio $E\left[\left(\tilde{Y} - Y\right)^2\right]$.

Para cada valor i-ésimo de la función en un $x_i$, el valor del error cuadrático en la aproximación será $\epsilon_i^2 = (y_i - \tilde{y}_i)^2$, y el objetivo será minimizar la sumatoria de todos los $\epsilon_i$, y donde $\tilde{y}_i = a + bx_i + cx_i^2$.
$$\epsilon_i^2 = \left(y_i - \tilde{y}_i\right)^2 = \left[y_i - \left(a + bx_i + cx_i^2\right)\right]^2$$

Con lo que se define la función de pérdida:
$$L\left(a, b, c\right) = \sum \left[y_i - \left(a + bx_i + cx_i^2\right)\right]^2$$

Ahora se necesita saber los valores críticos de $a$, $b$ y $c$ que puedan minimizar dicha función. Esto es encontrar los valores $a^*$, $b^*$ y $c^*$ tales que las derivadas parciales de $L$ respecto a cada uno de ellos sean iguales a 0.

$$\frac{\partial L\left(a^*, b^*, c^*\right)}{\partial a} = 0 \implies \frac{\partial L}{\partial a} = -2 \sum \left[y_i - \left(a^* + b^*x_i + c^*x_i^2\right)\right] = 0$$
$$\frac{\partial L\left(a^*, b^*, c^*\right)}{\partial b} = 0 \implies \frac{\partial L}{\partial b} = -2 \sum x_i\left[y_i - \left(a^* + b^*x_i + c^*x_i^2\right)\right] = 0$$
$$\frac{\partial L\left(a^*, b^*, c^*\right)}{\partial c} = 0 \implies \frac{\partial L}{\partial c} = -2 \sum x_i^2\left[y_i - \left(a^* + b^*x_i + c^*x_i^2\right)\right] = 0$$

Ahora eliminando el término `-2` (Se divide en ambos lados por `-2`) y distribuyendo la sumatoria llegamos a estas correspondientes ecuaciones:
$$\sum y_i - a n - b\sum x_i - c \sum x_i^2 = 0 \implies \sum y_i = a n + b\sum x_i + c \sum x_i^2$$
$$\sum x_iy_i - a \sum x_i - b\sum x_i^2 - c \sum x_i^3 = 0 \implies \sum x_iy_i = a \sum x_i + b\sum x_i^2 + c \sum x_i^3$$
$$\sum x_i^2y_i - a \sum x_i^2 - b\sum x_i^3 - c \sum x_i^4 = 0 \implies \sum x_i^2y_i = a \sum x_i^2 + b\sum x_i^3 + c \sum x_i^4$$

(Por sencillez en la escritura se quitó el superíndice $*$ en los coeficientes)

Podemos usar un script de Python para calcular todas las sumatorias de las ecuaciones.

In [ ]:
X = [0, 6, 10, 16, 21]
Y = [2, 72, 137, 271, 532]
n = 5

sum_x = 0
sum_y = 0
sum_x2 = 0
sum_xy = 0
sum_x3 = 0
sum_x2y = 0
sum_x4 = 0

for x, y in zip(X, Y):
    sum_x += x
    sum_y += y
    sum_x2 += x**2
    sum_xy += x*y
    sum_x3 += x**3
    sum_x2y += x**2*y
    sum_x4 += x**4

print("sum_x: ", sum_x)
print("sum_y: ", sum_y)
print("sum_x2: ", sum_x2)
print("sum_xy: ", sum_xy)
print("sum_x3: ", sum_x3)
print("sum_x2y: ", sum_x2y)
print("sum_x4: ", sum_x4)

sum_x:  53
sum_y:  1014
sum_x2:  833
sum_xy:  17310
sum_x3:  14573
sum_x2y:  320280
sum_x4:  271313


Con estos resultados podemos llegar al siguiente sistema de ecuaciones lineales:

$$\begin{cases}
&5a &+& 53b &+& 833c &=& 1014 \\
&53a &+& 833b &+& 14573c &=& 17310 \\
&833a &+& 14573b &+& 271313c &=& 320280 \\
\end{cases}$$

De aquí se pueden despejar rápidamente los coeficientes, sin embargo a continuación los obtenemos con un script de Python.

In [ ]:
import numpy as np

se_A = np.array([
    [n, sum_x, sum_x2],
    [sum_x, sum_x2, sum_x3],
    [sum_x2, sum_x3, sum_x4]
], dtype=float)
se_b = np.array([sum_y, sum_xy, sum_x2y], dtype=float)

coef = np.linalg.solve(se_A, se_b)

a = coef[0]
b = coef[1]
c = coef[2]

print("a: ", a)
print("b: ", b)
print("c: ", c)

a:  12.06861313868617
b:  0.14309002433087892
c:  1.135742092457422


Llegando así a que los estimadores de mínimos cuadrados para este modelo son:
$$a \approx 12.0686$$
$$b \approx 0.1431$$
$$c \approx 1.1357$$

### Ejercicio 3
(4 puntos) Don Francisco tiene 10 clientes a los que les ha vendido mercancías a crédito y, de ellos, 3 están en mora con el pago prometido. Matías, teniendo en cuenta la información disponible, considera que puede modelar el porcentaje $p$ de morosidad según una distribución $\Beta(1,4)$. Para determinar los parámetros $\alpha$ y $\beta$, decide usar inferencia bayesiana. Con esto, pretende explicarle a Don Francisco, cómo será el comportamiento de pago de sus clientes a crédito. Determinen la distribución a posteriori del parámetro pde porcentaje de morosidad ($\alpha$ y $\beta$). Determinar su media y su varianza.

Primero que nada, se recuerda la fórmula del teorema de Bayes generalizado para tenerla presente durante el desarrollo:
$$
f_{\Theta | \bar{X}=\bar{x}}(\theta) = \frac{f_{\bar{X}|\Theta=\theta}(\bar{x}) \pi(\theta)}{\int f_{\bar{X}|\Theta=\theta}(\bar{x}) \pi(\theta) d\theta}
$$
En este caso el parámetro $\theta$ será la tasa de morosidad $p$ del problema.
Para los datos observados del problema, por la naturaleza del mismo, podemos definir los parámetros para obtener una distribución binomial, con $k=3$ éxitos y $n-k=7$ fracasos, que represente la densidad condicional de los datos.
$$f_{\bar{X}|\Theta=\theta}(\bar{x}) = \binom{10}{3}p^3(1-p)^7$$

Respecto a la distribución a priori, se indica que es posible modelarla como una distribución beta con parámetros $\alpha=1$ y $\beta=4$. Esta es:
$$
\pi (\theta) = \frac{\theta^{\alpha - 1} (1 - \theta)^{\beta - 1}}{B(\alpha, \beta)}
$$
donde $\alpha$ y $\beta$ tienen los valores ya asignados y $B(\alpha, \beta)$ es la llamada función beta, que se define como una integral a calcular, que nos devolverá cierto valor y que, por el momento, se dejará representado de esa manera, sin avanzar en su cálculo.
$$
\pi (\theta) = \frac{\theta^{\alpha - 1} (1 - \theta)^{\beta - 1}}{B(\alpha, \beta)} = \frac{\theta^{1 - 1} (1 - \theta)^{4 - 1}}{B(1, 4)} = \frac{(1 - \theta)^{3}}{B(1, 4)}
$$
Reemplazando $\theta$ por el parámetro correspondiente es:
$$\pi (p) = \frac{(1 - p)^{3}}{B(1, 4)}$$

Teniendo la distribución a priori y la densidad condicional de los datos dado $\theta$, ahora podemos reemplazar en la fórmula del teorema generalizado de Bayes y resolver para obtener la distribución a posteriori.
$$
f_{\Theta | \bar{X}=\bar{x}}(\theta) = \frac{\binom{10}{3}p^3(1-p)^7 \cdot \frac{(1 - p)^{3}}{B(1, 4)}}{\int \binom{10}{3}p^3(1-p)^7 \cdot \frac{(1 - p)^{3}}{B(1, 4)} dp}
$$
Los términos $\binom{10}{3}$ y $B(1, 4)$ pueden sacarse de la integral del denominador y simplificarse con los mismos términos del numerador, tal que luego, juntando los términos $(1 - p)$ (sumando sus exponentes) en numerador y denominador, llegamos a:
$$
f_{P | \bar{X}=\bar{x}}(p) = \frac{\binom{10}{3}p^3(1-p)^7 \cdot \frac{(1 - p)^{3}}{B(1, 4)}}{\int \binom{10}{3}p^3(1-p)^7 \cdot \frac{(1 - p)^{3}}{B(1, 4)} dp} = \frac{p^3(1-p)^{10} }{\int p^3(1-p)^{10}  dp}
$$

Por otro lado, conocemos que la distribución conjugada correspondiente es una Beta, que como, ya sabemos de la a priori, tiene la forma:
$$
\Beta (\alpha', \beta') = \frac{x^{\alpha' - 1} (1 - x)^{\beta' - 1}}{\int x^{\alpha' - 1} (1 - x)^{\beta' - 1} dx}
$$

Comparando con la distribución a posteriori calculada, podemos ver que que efectivamente tienen un forma similar y de esta podemos igualar los exponentes:
$$\alpha' - 1 = 3 \implies \alpha' = 4$$
$$\beta' - 1 = 10 \implies \beta' = 11$$

Finalmente, podemos decir que la distribución a posteriori del porcentaje de morosidad es una distribución beta $\Beta(4, 11)$, con media $\mu$ y varianza $\sigma^2$:
$$\mu = \frac{\alpha'}{\alpha' + \beta'} = \frac{4}{4 + 11} \approx 0.2667$$
$$\sigma^2 = \frac{\alpha' \beta'}{(\alpha' + \beta')^2 (\alpha' + \beta' + 1)} = \frac{4\cdot 11}{(4 + 11)^2 (4 + 11 + 1)} \approx 0.0122$$